vital/algo/resection_camera.h

```diff
@@ -0,0 +1,100 @@
+// This file is part of KWIVER, and is distributed under the
+// OSI-approved BSD 3-Clause License. See top-level LICENSE file or
+// https://github.com/Kitware/kwiver/blob/master/LICENSE for details.
+
+/// \file
+/// \brief resection_camera algorithm definition
+
+#ifndef VITAL_ALGO_RESECTION_CAMERA_H_
+#define VITAL_ALGO_RESECTION_CAMERA_H_
+
+#include <vital/algo/algorithm.h>
+#include <vital/types/camera_perspective.h>
+#include <vital/types/feature_track_set.h>
+#include <vital/types/landmark_map.h>
+
+#include <vector>
+
+namespace kwiver {
+
+namespace vital {
+
+namespace algo {
+
+/// An abstract base class to resection a camera using 3D feature and point
+/// projection pairs.
+class VITAL_ALGO_EXPORT resection_camera
+  : public kwiver::vital::algorithm_def< resection_camera >
+{
+public:
+  /// \return name of this algorithm
+  static std::string
+  static_type_name() { return "resection_camera"; }
+
+  /// Estimate camera parameters from 3D points and their corresponding
+  /// projections.
+  ///
+  /// \param [in] pts2d 2d projections of pts3d points
+  /// \param [in] pts3d 3d points in a 1-1 correspondence with pts2d
```

Comment:
```suggestion
  /// \param [in] image_points 2D
  ///   points in camera image space to use for registration
  /// \param [in] world_points
  ///   locations in world space corresponding to the \p image_points
```

I'm not sure it's needed, but we might also add:

```c++
  /// \note The \p image_points and \p world_points must have a one-to-one
  ///       correspondence.
```